### PR TITLE
Create tests for Experience servlet

### PR DESCRIPTION
**Pull request by lattai**
Will fill in the doGet and doPost next.

### CODE REVIEW

File: src/test/java/com/google/sps/ExperienceServletTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  @Rule public ExpectedException emptyImageRule = ExpectedException.none();
+
+  @Test
+  public void doGet_successfulPath_getsPlayerExp() throws IOException, LoggedOutException {
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(this.response.getWriter()).thenReturn(printWriter);
+    this.experienceServlet.doGet(this.request, this.response);
+    // int result = Integer.parseInt(stringWriter.toString());
+    Assert.assertTrue(true);
```
Comment (lzoebaker) [inline src/test/java/com/google/sps/ExperienceServletTest.java:102]: No need to change it now, but for next time, with non implemented methods, you could follow the design pattern that Will/Dylan showed up and throw a custom NotImplementedException() in your stub method, then have your test method assert that its being thrown.

Comment (lattai) [reply src/test/java/com/google/sps/ExperienceServletTest.java:102]: Gotcha! I'll do that next time

File: src/main/java/com/google/sps/servlets/ExperienceServlet.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  public void init() {
+    this.gson = new Gson();
+    this.userService = UserServiceFactory.getUserService();
+    this.datastore = DatastoreServiceFactory.getDatastoreService();
+    this.playerDatabase = new PlayerDatabase(datastore, userService);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Update the player's experience points
```
Comment (WillHipschman) [inline src/main/java/com/google/sps/servlets/ExperienceServlet.java:34]: probably throw UnsupportedOperation and add TODOs for both these methods so it's clear they are still in progress.

File: src/test/java/com/google/sps/ExperienceServletTest.java (hunk 1 above)
Comment (WillHipschman) [reply src/test/java/com/google/sps/ExperienceServletTest.java:102]: Let's do that this time :)

File: src/test/java/com/google/sps/ExperienceServletTest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    // Assert.assertTrue(result == TEST_EXPERIENCE);
```
Comment (WillHipschman) [inline src/test/java/com/google/sps/ExperienceServletTest.java:103]: Don't check in commented out code

File: src/test/java/com/google/sps/ExperienceServletTest.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    when(this.request.getParameter(EXPERIENCE_PARAMETER)).thenReturn(testExperiencePlusOne);
+    this.experienceServlet.doPost(this.request, this.response);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(this.response.getWriter()).thenReturn(printWriter);
+    this.experienceServlet.doGet(this.request, this.response);
+    // int result = Integer.parseInt(stringWriter.toString());
+    Assert.assertTrue(true);
+    // Assert.assertTrue(result == (TEST_EXPERIENCE + 1));
```
Comment (WillHipschman) [inline src/test/java/com/google/sps/ExperienceServletTest.java:121]: here too

File: src/test/java/com/google/sps/ExperienceServletTest.java (hunk 1 above)
Comment (DirtGrubDylan) [reply src/test/java/com/google/sps/ExperienceServletTest.java:102]: +1

File: src/test/java/com/google/sps/ExperienceServletTest.java (hunk 3 above)
Comment (DirtGrubDylan) [reply src/test/java/com/google/sps/ExperienceServletTest.java:103]: +1

**Pull request merged**